src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/CompilationHelper.cs

```diff
@@ -22,27 +23,30 @@ public class CompilationHelper
             // workaround https://github.com/dotnet/roslyn/pull/55866. We can remove "LangVersion=Preview" when we get a Roslyn build with that change.
             .WithLanguageVersion(LanguageVersion.Preview);
 
+#if NETCOREAPP
+        private static readonly Assembly systemRuntimeAssembly = System.Runtime.Loader.AssemblyLoadContext.Default.LoadFromAssemblyName(new AssemblyName("System.Runtime"));
+#endif
+
         public static Compilation CreateCompilation(
             string source,
             MetadataReference[] additionalReferences = null,
             string assemblyName = "TestAssembly",
             bool includeSTJ = true)
         {
-            // Bypass System.Runtime error.
-            Assembly systemRuntimeAssembly = Assembly.Load("System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
-            Assembly systemCollectionsAssembly = Assembly.Load("System.Collections, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
-            string systemRuntimeAssemblyPath = systemRuntimeAssembly.Location;
-            string systemCollectionsAssemblyPath = systemCollectionsAssembly.Location;
 
             List<MetadataReference> references = new List<MetadataReference> {
                 MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
                 MetadataReference.CreateFromFile(typeof(Attribute).Assembly.Location),
                 MetadataReference.CreateFromFile(typeof(Type).Assembly.Location),
-                MetadataReference.CreateFromFile(typeof(KeyValuePair).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(KeyValuePair<,>).Assembly.Location),
                 MetadataReference.CreateFromFile(typeof(ContractNamespaceAttribute).Assembly.Location),
                 MetadataReference.CreateFromFile(typeof(JavaScriptEncoder).Assembly.Location),
-                MetadataReference.CreateFromFile(systemRuntimeAssemblyPath),
-                MetadataReference.CreateFromFile(systemCollectionsAssemblyPath),
+                MetadataReference.CreateFromFile(typeof(GeneratedCodeAttribute).Assembly.Location),
+                MetadataReference.CreateFromFile(typeof(ReadOnlySpan<>).Assembly.Location),
+#if NETCOREAPP
+                MetadataReference.CreateFromFile(typeof(LinkedList<>).Assembly.Location),
+                MetadataReference.CreateFromFile(systemRuntimeAssembly.Location),
```

Comment:
```suggestion
```
I do not think you need this line for netcoreapp.

You have `MetadataReference.CreateFromFile(typeof(object).Assembly.Location),` in the list that will resolve to CoreLib. `System.Runtime` is a subset of what CoreLib has.

Reply:
Judging by older comments, I'm guessing Roslyn was complaining because it didn't have a reference to `System.Runtime`? I guess we can remove it altogether and see what happens.

Reply:
Confirmed that Roslyn is failing if System.Runtime is not specified. So we need to keep this.